airflow/providers/common/sql/hooks/sql.py

```diff
@@ -264,22 +291,26 @@ def run(
 
             with closing(conn.cursor()) as cur:
                 results = []
-                for sql_statement in sql:
+                for sql_statement in sql_list:
                     self._run_command(cur, sql_statement, parameters)
 
                     if handler is not None:
                         result = handler(cur)
-                        results.append(result)
-                self.last_description = cur.description
+                        if has_scalar_return_value(sql, return_last, split_statements):
+                            results = [result]
```

Comment:
We do not need to append value - only last is used anyway when scalar return value is used.

Reply:
I have a better idea.

Reply:
ok. I solved it much better now